protect-api/README.md

```diff
@@ -0,0 +1,113 @@
+| **Name**                       | `java-api`                                                           | Suggested value for this sample. <br/> You can change the app name at any time.  |
+| **Supported account types**    | **Accounts in this organizational directory only (Single tenant)**   | Suggested value for this sample.                                                 |
+| **Platform type**              | _None_                                                               | No redirect URI required; don't select a platform.                               |
+| **Scopes defined by this API** | **Scope name**: `Greeting.Read`<br/>**Who can consent?**: **Admins and users**<br/>**Admin consent display name**: `Read API Greetings`<br/>**Admin consent description**: `Allows the user to see greetings from the API.`<br/>**User consent display name**: `Read API Greetings`<br/>**User consent description**: `Allows you to see greetings from the API.`<br/>**State**: **Enabled** | Required scope for this sample. |
+
+> :information_source: **Bold text** in the tables above matches (or is similar to) a UI element in the Azure portal, while `code formatting` indicates a value you enter into a text box in the Azure portal.
+
+### 2. Update code sample with app registration values
+
+Open the [_application.yml_](src/main/resources/application.yml) file and modify the three Azure Active Directory configuration properties using the values from your [app's registration in the Azure portal](https://docs.microsoft.com/azure/active-directory/develop/quickstart-register-app).
+
+```yaml
+# 'Tenant ID' of your Azure AD instance - this value is a GUID
+tenant-id: value-here
+
+# 'Application ID URI' of app registration in Azure portal - this value typically starts with api://
+app-id-uri: value-here
+```
+
+### 3. Install package(s) and compile application
+
+To install Spring Boot and MSAL libraries:
+
+```bash
+mvn install
+```
+
+## Run the application
+
+```bash
+mvn spring-boot:run
+```
+
+## Access the api
+
+Open postman, curl, or similar and make an HTTP GET request to **http://localhost:8080** with an `Authorization` header of `Bearer {valid-access-token}`. If everything worked, the sample app should produce output similar to this:
+
+```console
+$ curl http://localhost:8080 -H "Authorization: Bearer {valid-access-token}"
+Hello, world. You were able to access this because you provided a valid access token with the Greeting.Read scope as a claim.
+```
+
+## About the code
+
+This Java API uses the Spring Boot web framework. The app has a single route that requires an access token. The access token will be automatically validated by MSAL. A missing or invalid (expired, wrong audience, etc) token will result in a `401` response. A otherwise valid token without the proper scopes will result in a `403` response. A valid token with a proper scope of (`Greeting.Read`) will result in the "Hello, world." message.

```

Comment:
```suggestion
This Java API uses the Spring Boot web framework. The app has a single route that requires an access token. The access token will be automatically validated by MSAL:

- A missing or invalid (expired, wrong audience, etc) token will result in a `401` response.
- An otherwise valid token without the proper scope will result in a `403` response.
- A valid token with the proper scope of `Greeting.Read` will result in the "Hello, world." message.
```